File: functional-exercises/exercises.erl
-module(exercises).

-export([existe/2, maxlis/1]).

existe(Lis, E) ->
    Xs = lists:map(fun(X) -> X == E end, Lis),
    lists:foldr(fun(X, Acc) -> X or Acc end, false, Xs).

maxn(X, Acc) ->
    if X > Acc ->
	X;
    true ->
        Acc
    end.

maxlis(Lis) ->
    [X|Xs] = Lis,
    lists:foldr(fun maxn/2, X, Xs).

File: functional-exercises/exercises.clj
(defn existe [lis x]
 (reduce (fn [a b] (or a b)) (map (fn [y] (= x y)) lis))) 

(defn maxn [a b]
 (if (> a b)
  a
  b))

(defn maxlis [lis]
 (reduce maxn lis))

(defn base10 [x y]
  (* x (Math/pow 10 y)))

(defn dignum [digs]
  (let [dig_max (count digs)
        powers (reverse (range dig_max))]
      (reduce + (map base10 digs powers))))
